src/routes/organizations/entities/user-organization-role.entity.ts

```diff
@@ -0,0 +1,15 @@
+import { UserOrganizationRole } from '@/domain/users/entities/user-organization.entity';
+import { z } from 'zod';
+
+function isUserOrganizationRole(
+  role: string,
+): role is keyof typeof UserOrganizationRole {
+  return role in UserOrganizationRole;
+}
+
+export const UserOrganizationRoleSchema = z
+  .enum(['ADMIN', 'MEMBER'])
+  .refine(isUserOrganizationRole)
+  .transform((role) => {
+    return UserOrganizationRole[role];
+  });
```

Comment:
As per an associated discussion, this should be made generic for other areas that require string <> numeric enums.

Reply:
As we now transform on the entity level, this can be disregarded. We can use `z.enum(getStringEnumKeys(...))` instead.